src/CodeStyle/CSharp/Analyzers/Microsoft.CodeAnalysis.CSharp.CodeStyle.csproj

```diff
@@ -19,11 +19,31 @@
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\ArgumentSyntaxExtensions.cs" Link="Extensions\ArgumentSyntaxExtensions.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\AssignmentExpressionSyntaxExtensions.cs" Link="Extensions\AssignmentExpressionSyntaxExtensions.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\ContextQuery\SyntaxTokenExtensions_SharedWithCodeStyle.cs" Link="Extensions\ContextQuery\SyntaxTokenExtensions_SharedWithCodeStyle.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\ContextQuery\SyntaxTreeExtensions_Shared.cs" Link="Extensions\ContextQuery\SyntaxTreeExtensions_Shared.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\DirectiveSyntaxExtensions.cs" Link="Extensions\DirectiveSyntaxExtensions.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\DirectiveSyntaxExtensions.DirectiveInfo.cs" Link="Extensions\DirectiveSyntaxExtensions.DirectiveInfo.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\DirectiveSyntaxExtensions.DirectiveSyntaxEqualityComparer.cs" Link="Extensions\DirectiveSyntaxExtensions.DirectiveSyntaxEqualityComparer.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\DirectiveSyntaxExtensions.DirectiveWalker.cs" Link="Extensions\DirectiveSyntaxExtensions.DirectiveWalker.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\ExpressionSyntaxExtensions_Shared.cs" Link="Extensions\ExpressionSyntaxExtensions_Shared.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs" Link="Extensions\ITypeSymbolExtensions.TypeSyntaxGeneratorVisitor.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\ITypeSymbolExtensions_Shared.cs" Link="Extensions\ITypeSymbolExtensions_Shared.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\MemberDeclarationSyntaxExtensions_GetAttributes.cs" Link="Extensions\MemberDeclarationSyntaxExtensions_GetAttributes.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\MemberDeclarationSyntaxExtensions_Shared.cs" Link="Extensions\MemberDeclarationSyntaxExtensions_Shared.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\StatementSyntaxExtensions.cs" Link="Extensions\StatementSyntaxExtensions.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\StringExtensions.cs" Link="Extensions\StringExtensions.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\SyntaxListExtensions.cs" Link="Extensions\SyntaxListExtensions.cs" />
+    <Compile Include="..\..\..\Workspaces\CSharp\Portable\Extensions\SyntaxNodeExtensions_SharedWithCodeStyle.cs" Link="Extensions\SyntaxNodeExtensions_SharedWithCodeStyle.cs" />
```

Comment:
These will certainly clash with my refactoring PR as I have split and moved bunch of these into the shared layer as well. One of us will have to deal with merge conflicts in few of files being added to shared layer now.

Reply:
Yes.  I think that is def the case.  But ideally you shouldn't need to split/move these down *yourself*.  the only splitting/moving would be for some extensions that weren't captured in this that are used by some other features?

but in tha tcase, i would argue again that they shouldn't be in that big PR.  we should pull them out (like the unnecessary-imports part) so that none of that is in the big PR).

Reply:
> but in tha tcase, i would argue again that they shouldn't be in that big PR. we should pull them out 

The refactoring PR is now just doing this exact thing: just splitting and moving down extensions into shared layers. Your PR here is doing so for SyntaxGenerator and SyntaxFacts and my PR is doing so for other features. I think the key is we should avoid these efforts in parallel as there will be overlap that leads to big merge conflicts. I would recommend waiting on this PR till the primary refactoring PR goes in.

Reply:
Note: it may be hte case that that's what is going on in that PR... but it's also 8k more adds and 3k more deletes.  So it's mixed in with a *bunch* of other stuff.  We might want to consider how this could just go in, and then have the remaining stuff be teh final bit we need to review in isolation.